src/content/docs/en/guides/typescript.mdx

```diff
@@ -214,6 +214,22 @@ type MyAttributes = astroHTML.JSX.ImgHTMLAttributes;
 ```
 :::
 
+### Component Props type
+
+<p><Since v="4.3.0" /></p>
+
+In order to reference the type of `Props` accepted by another component, Astro exposes the `ComponentProps` utility from `astro/types`.
+
+```astro
```

Comment:
``````suggestion
### `ComponentProps` type

<p><Since v="4.3.0" /></p>

This type export allows users to reference the `Props` accepted by another component, even if that component doesn't export that `Props` type directly.

The following example shows using the `ComponentProps` utility from `astro/types` to reference a `<Button />` component's `Props` types:

```astro title="src/pages/index.astro"
``````
Would something like this be correct, @natemoo-re ? Please fine-tune as necessary. I really liked what you wrote in the PR description as an intro!

Reply:
Love this suggestion as-is, thanks!